### PR TITLE
update documentation to include kuberentes initContainers

### PR DESCRIPTION
**Pull request by mickfeech**
##### SUMMARY

Updating documentation to include details around using initContainers instead up creating a custom image

##### ISSUE TYPE
Docs Pull Request

##### COMPONENT NAME
Docs

##### AWX VERSION
```
make: Nothing to be done for `version'.
```

### CODE REVIEW

**Review by ryanpetrello — APPROVED** [review]
@mickfeech thanks for putting this together, I like it :)

File: docs/custom_virtualenvs.md [hunk 1]
```diff
@@ -77,6 +77,48 @@ index d69e2c9..a08bae5 100644
...
+
+    volumes:
+        - emptyDir: {}
+          name: custom-venv
+
+Now create an initContainer stanza.  You can subsititute your own custom images for this example we are using centos:7 as the base to build upon.  The command stanza is where you will add the python modules you require in your virtualenv.
+
+    initContainers:
+        - image: 'centos:7'
+        - name: init-my-custom-venv
```
Comment (matburt) [inline docs/custom_virtualenvs.md]: The extra dashes here are going to cause problems, I think it should look like:

```
initContainers:
   - image: ...
     name: ....
     command:
        - sh
        - ....
```

**Review by matburt — CHANGES_REQUESTED** [review]

File: docs/custom_virtualenvs.md [hunk 2]
```diff
@@ -77,6 +77,48 @@ index d69e2c9..a08bae5 100644
...
+              virtualenv /var/lib/awx/venv/my-custom-venv &&
+              source /var/lib/awx/venv/my-custom-venv/bin/activate &&
+              /var/lib/awx/venv/my-custom-venv/bin/pip install python-memcached psutil &&
+              /var/lib/awx/venv/my-custom-venv/bin/pip install -U "ansible == X.Y.Z" &&
+              /var/lib/awx/venv/my-custom-venv/bin/pip install -U custom-python-module
+        - volumeMounts:
+            - mountPath: /var/lib/awx/venv/my-custom-venv
+              name: custom-venv
+
+Fiinally in the awx-celery container stanza add the shared volume as a mount.
```
Comment (matburt) [inline docs/custom_virtualenvs.md]: This is probably going to need to appear in the `awx-web` container also in order for it to appear as a selectable venv from the web service for `/api/v2/config`

**Comment by matburt** [comment]
This is a really excellent doc, and is a great alternative to a full rebuild of the image. I'd like to promote this to the general user documentation also.

**Comment by ryanpetrello** [comment]
@mickfeech would you mind rebasing against `devel` so we can merge this?

**Comment by matburt** [comment]
I've removed the `mergeit` label because I had some feedback above that needs to be addressed and it looks it might still need a rebase.  If we can take care of these I'll reapply the mergeit label.

File: docs/custom_virtualenvs.md (hunk 2 above)
Comment (mickfeech) [reply docs/custom_virtualenvs.md]: You're right.  Just updated.

File: docs/custom_virtualenvs.md (hunk 1 above)
Comment (mickfeech) [reply docs/custom_virtualenvs.md]: Correct.  Updated.

**Comment by sharkcus** [comment]
Attempted with version 3.0.1 and I was required to mount the volume in the awx-web container in addition to the awx-celery container before it would recognize the venv in the api. This did not seem to be required in a 1.0.8 instance. Can someone else confirm?

**Comment by matburt** [comment]
@sharkcus it does say that in the documentation provided in this PR

> Fiinally in the awx-celery and awx-web containers stanza add the shared volume as a mount.

File: docs/custom_virtualenvs.md [hunk 3]
```diff
@@ -77,6 +77,48 @@ index d69e2c9..a08bae5 100644
...
+              virtualenv /var/lib/awx/venv/my-custom-venv &&
+              source /var/lib/awx/venv/my-custom-venv/bin/activate &&
+              /var/lib/awx/venv/my-custom-venv/bin/pip install python-memcached psutil &&
+              /var/lib/awx/venv/my-custom-venv/bin/pip install -U "ansible == X.Y.Z" &&
+              /var/lib/awx/venv/my-custom-venv/bin/pip install -U custom-python-module
+          volumeMounts:
+            - mountPath: /var/lib/awx/venv/my-custom-venv
+              name: custom-venv
+
+Fiinally in the awx-celery and awx-web containers stanza add the shared volume as a mount.
```
Comment (matburt) [inline docs/custom_virtualenvs.md]: Fiinally -> Finally @mickfeech if you could fix that.

**Review by matburt — APPROVED** [review]

**Pull request merged**